Source/Core/IdMap.cs

```diff
@@ -0,0 +1,28 @@
+using System.Collections.Generic;
+
+namespace Microsoft.Boogie
+{
+  public class IdMap<T>
+  {
+    Dictionary<T, int> toId = new ();
+    Dictionary<int, T> fromId = new ();
+    private int counter = 10;
```

Comment:
Why 10?

Reply:
I don't have a good answer to that. `0` doesn't work, it seems like that's a reserved number in the control flow scheme but I didn't investigate to confirm that.

I'll change the code to explain the choice for 10.

Reply:
Ok, I remember now.  0 is indeed reserved because it is the value assigned to the fictitious control-flow location at the beginning of an error trace.  Please include this note as a comment.

Reply:
Done